brian2genn/device.py

```diff
@@ -457,7 +477,17 @@ def code_object(self, owner, name, abstract_code, variables, template_name,
                                         variables=variables,
                                         variable_indices=variable_indices,
                                         override_conditional_write=override_conditional_write)
+            # We need to clear the array cache for these at some point (normally
+            # would be done in cpp_standalone.device.code_object()
+            # I will do it here but not sure this is the best place
+            # I am also not sure whether "written_readonly_vars" apply here
+            # I WILL ASSUME NOT
+            for var in codeobj.variables.values():
+                if (isinstance(var, ArrayVariable) and
+                    not var.read_only):
+                    self.array_cache[var] = None
```

Comment:
I have to look into this a bit more in detail before I can give you a clear :+1: / :-1: